fixtures-code/process-compose-shutdown-inorder.yaml

```
@@ -0,0 +1,31 @@
+version: "0.5"
+
+log_level: debug
+log_length: 1000
+
+processes:
+  procA:
+    command: |
+      echo "A: starting"
+
+  procB:
+    command: |
+      echo "B: starting"
+    is_daemon: true
+    shutdown:
+      command: |
+        echo "B: exiting"
+    depends_on:
+      procA:
+        condition: process_started
+
+  procC:
+    command: |
+      echo "C: starting"
+    isDaemon: true
+    shutdown:
+      command: |
+        echo "C: exiting"
```

You could write these to a file, then assert at the end for the contents to be ordered backwards.
```suggestion
        echo "C: exiting" >> shutdown-inorder.log
        sleep 1
```

---

Another approach:
https://github.com/F1bonacc1/process-compose/commit/9ac79177aeed4fb471faa644fbe4ccf9ad9a7360

---

Nice, thanks. I cherry-picked it to this branch.